presto-hive/src/test/java/com/facebook/presto/hive/TestGenericHiveRecordCursorProvider.java

```diff
@@ -60,44 +73,43 @@
     public static final HiveColumnHandle ID = new HiveColumnHandle("id", HIVE_STRING, parseTypeSignature(StandardTypes.VARCHAR), 5, REGULAR, Optional.empty(), Optional.empty());
     public static final HiveColumnHandle LAST_UPDATE_MONTH = new HiveColumnHandle("last_update_month", HIVE_STRING, parseTypeSignature(StandardTypes.VARCHAR), 6, REGULAR, Optional.empty(), Optional.empty());
     public static final HiveColumnHandle LAST_UPDATE_TIME = new HiveColumnHandle("last_update_time", HIVE_STRING, parseTypeSignature(StandardTypes.VARCHAR), 7, REGULAR, Optional.empty(), Optional.empty());
-    public static final String CUSTOM_SPLIT_CLASS = "custom_split_class";
-    public static final String HUDI_DELTA_FILEPATHS = "hudi_delta_filepaths";
-    public static final String HUDI_BASEPATH = "hudi_basepath";
-    public static final String HUDI_MAX_COMMIT_TIME = "hudi_max_commit_time";
     public static final String TABLE_NAME = "hudi_part_mor_rt";
     public static final String FILE_NAME = "b3711ddf-8c11-4666-82ec-fbc952e1dc72-0_1-61-24052_20210524095413.parquet";
 
     @Test
-    public void shouldReturnHudiRecordCursorWithCopyOnFirstWriteEnabled()
+    public void shouldReturnHudiRecordCursorForRealtimeFileSplitWithCopyOnFirstWriteEnabled()
     {
-        Optional<RecordCursor> recordCursor = getRecordCursor(true);
+        Optional<RecordCursor> recordCursor = getRecordCursor(HoodieRealtimeFileSplit.class, true);
         assertTrue(recordCursor.isPresent());
     }
 
     @Test
-    public void shouldReturnHudiRecordCursorWithCopyOnFirstWriteDisabled()
+    public void shouldReturnHudiRecordCursorForRealtimeFileSplitWithCopyOnFirstWriteDisabled()
     {
-        Optional<RecordCursor> recordCursor = getRecordCursor(false);
+        Optional<RecordCursor> recordCursor = getRecordCursor(HoodieRealtimeFileSplit.class, false);
         assertTrue(recordCursor.isPresent());
     }
 
-    private static Optional<RecordCursor> getRecordCursor(Boolean isCopyOnFirstWriteConfigurationEnabled)
+    @Test
+    public void shouldReturnHudiRecordCursorForRealtimeBootstrapBaseFileSplitWithCopyOnFirstWriteEnabled()
+    {
+        Optional<RecordCursor> recordCursor = getRecordCursor(HoodieRealtimeBootstrapBaseFileSplit.class, true);
+        assertTrue(recordCursor.isPresent());
+    }
+
+    @Test
+    public void shouldReturnHudiRecordCursorForRealtimeBootstrapBaseFileSplitWithCopyOnFirstWriteDisabled()
+    {
+        Optional<RecordCursor> recordCursor = getRecordCursor(HoodieRealtimeBootstrapBaseFileSplit.class, false);
+        assertTrue(recordCursor.isPresent());
```

Comment:
Can we try to verify that the fileSplit behind this record cursor actually has BootstrapSplit? 

The reason I am thinking this is recordCursor was still present even before your changes.

Reply:
Changes in commit 9dcdc46 are similar to the changes done here #19096 for MOR RT tables and these tests(added in 19096 only) just extend them for MOR bootstrap tables.
These test changes are not for the other commit that I have done in this PR.
I could not find any way to extract fileSplit from the record cursor.

Reply:
Thank you for confirming.